### PR TITLE
Bug 1132418 - [FTE] Incorrect coloured status bar and icons

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=78a447c570b4)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=a86c249bf081)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4a55f11c65b5)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8b5fae9b100a)

File: apps/system/js/app_chrome.js [hunk 1]
```diff
@@ -503,12 +503,18 @@
...
   AppChrome.prototype.setThemeColor = function ac_setThemColor(color) {
-
     // Do not set theme color for private windows
     if (this.app.isPrivateBrowser()) {
       return;
     }
 
+    if (this.app.CLASS_NAME === 'PopupWindow' &&
+      this.app.rearWindow &&
+      this.app.rearWindow.themeColor) {
```
Comment (etiennesegonzac) [inline apps/system/js/app_chrome.js]: should we use `getBottomMostWindow()` ?

File: apps/system/js/popup_window.js [hunk 2]
```diff
@@ -21,6 +21,19 @@
     };
 
     AppWindow.call(this, configs);
+
+    // Replicate the theme color from the parent window.
+    // See http://bugzil.la/1132418
+    if (this.rearWindow) {
+      this.themeColor = this.rearWindow.themeColor;
+      this.element.classList.toggle('light',
+        this.rearWindow.element.classList.contains('light'));
```
Comment (etiennesegonzac) [inline apps/system/js/popup_window.js]: should we use `AppChrome.prototype.useLightTheming()`?

File: apps/system/js/popup_window.js [hunk 3]
```diff
@@ -21,6 +21,19 @@
...
+
+      if (this.statusbar) {
```
Comment (etiennesegonzac) [inline apps/system/js/popup_window.js]: looks like this is the only place where we style `statusbar.titleBar` directly, I have a bad feeling about this.
can you explain a little more?

File: apps/system/js/popup_window.js [hunk 4]
```diff
@@ -21,6 +21,19 @@
     };
 
     AppWindow.call(this, configs);
+
+    // Replicate the theme color from the parent window.
+    // See http://bugzil.la/1132418
+    if (this.rearWindow) {
```
Comment (etiennesegonzac) [inline apps/system/js/popup_window.js]: wouldn't this always be true for a `popupWindow`?

**Comment by etiennesegonzac** [comment]
Node: @KevinGrandon is trying to remove the dev_app folder, maybe the content of the theme-color app should be moved to the uitest app.

File: apps/system/js/popup_window.js (hunk 4 above)
Comment (gmarty) [inline apps/system/js/popup_window.js]: A few lines above, the existence of `configs.rearWindow` is tested, so I thought it makes sense to test `this.rearWindow` too. But I can remove it if you think it's useless.

File: apps/system/js/app_chrome.js (hunk 1 above)
Comment (gmarty) [inline apps/system/js/app_chrome.js]: Yep and it's more elegant.

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ad395e379518)

File: apps/system/js/popup_window.js (hunk 2 above)
Comment (gmarty) [inline apps/system/js/popup_window.js]: Yep, that's correct!

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=179889b779a9)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2d3e0c6c1656)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=24f062922a82)

**Comment by KevinGrandon** [comment]
Yes - I would prefer to place the app in this folder, and load it when needed: https://github.com/mozilla-b2g/gaia/tree/master/apps/system/test/marionette

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=eb0f08737ab4)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=a8c023e70e27)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=638471557c05)

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8bcb821aa930)

File: apps/system/js/popup_window.js (hunk 4 above)
Comment (etiennesegonzac) [inline apps/system/js/popup_window.js]: let's keep it but make it an early return.
```
if (!this.rearWindow) {
  return;
}
[...]
```

**Comment by try-server-hook** [comment]
[gmarty Guillaume C. Marty (gmarty)](https://github.com/gmarty) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d56f1d6c4d36)

**Pull request merged**